### PR TITLE
Added argparser, fixed exception handling, removed unnecessary modules

### PR DESCRIPTION
**Pull request by dimsanius**
I am currently working on slightly different implementation of VM snapshotter via Python. However, I am still using this project as foundation. So, here is my contribution to this project. (Thanks for making the code public btw)

Following changes were made:

- Introduced `argparser` - now help menu can be accessed (using `-h` or `--help`) and no annoying `check_arguments` method is needed as argparser will perform checks by itself. The method of passing in VM name remains the same: `python main.py "Awesome VM Name"`
- Introduced `try...except` statements within `delete_last_snapshot` and `create_snapshot` in case non-existing VM name is provided (or any other exception is risen)
- Cleansed unused module imports

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,62 +1,44 @@
...
+            print("Deleted " + first_snapshot[1])
+    except Exception as ex:
+        print(f"Snapshot deletion exited prematurely with following exception: {ex}")
+        return
 
-def create_snapshot(machine_name=None):
+def create_snapshot(machine_name):
     vm_initial_status = 1  # zero means powered on
-
-    if machine_name:
```
Comment (dimsanius) [inline main.py:59]: Machine name check is removed as it is now handled by `argparser`

File: main.py [hunk 2]
```diff
@@ -1,62 +1,44 @@
...
-        print("Not enough argument provided")
-        return 1
-    elif not isinstance(sys.argv[1], str):
-        print("Provided name is not istance of string")
-        return 1
-    return sys.argv[1]
-
-def delete_last_snapshot(machine_name=None):
-
-    if machine_name:
```
Comment (dimsanius) [inline main.py:23]: Machine name check is removed as it is now handled by `argparser`

File: main.py [hunk 3]
```diff
@@ -84,19 +66,37 @@ def create_snapshot(machine_name=None):
         if vm_initial_status:
             if session.state == virtualbox.library.SessionState(2):
                 session.unlock_machine()
-    else:
-        print("machine_name is None")
+    except Exception as ex:
+        print(f"Snapshot creation exited prematurely with following exception: {ex}")
+        # This will skip power down of machine on failure
+        return 1
```
Comment (dimsanius) [inline main.py:72]: Since exception is generic, the nature of exception is hard to determine. Therefore, returning 1 to prevent further action to a VM.

**Comment by mattiabaldari** [comment]
@dimsanius thanks a lot for the contribution! I really appreciate it

**Pull request merged**